code/modules/antagonists/eldritch_cult/eldritch_magic.dm

```diff
@@ -0,0 +1,395 @@
+/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash
+	name = "Ashen passage"
+	desc = "Low range spell allowing you to pass through a few walls."
+	school = "transmutation"
+	invocation = "ASH'N P'SSG'"
+	invocation_type = "whisper"
+	charge_max = 150
+	range = -1
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+	jaunt_in_time = 13
+	jaunt_duration = 10
+	jaunt_in_type = /obj/effect/temp_visual/dir_setting/ash_shift
+	jaunt_out_type = /obj/effect/temp_visual/dir_setting/ash_shift/out
+
+/obj/effect/proc_holder/spell/targeted/ethereal_jaunt/shift/ash/long
+	jaunt_duration = 50
+
+/obj/effect/temp_visual/dir_setting/ash_shift
+	name = "ash_shift"
+	icon = 'icons/mob/mob.dmi'
+	icon_state = "ash_shift2"
+	duration = 13
+
+/obj/effect/temp_visual/dir_setting/ash_shift/out
+	icon_state = "ash_shift"
+
+/obj/effect/proc_holder/spell/targeted/touch/mansus_grasp
+	name = "Mansus Grasp"
+	desc = "Touch spell that let's you channel the power of the old gods through you."
+	hand_path = /obj/item/melee/touch_attack/mansus_fist
+	school = "evocation"
+	charge_max = 150
+	clothes_req = FALSE
+	action_icon = 'icons/mob/actions/actions_ecult.dmi'
+	action_icon_state = "ash_shift"
+	action_background_icon_state = "bg_ecult"
+
+/obj/item/melee/touch_attack/mansus_fist
+	name = "Mansus Grasp"
+	desc = "A sinister looking aura that distorts the flow of reality around it."
+	icon_state = "disintegrate"
+	item_state = "disintegrate"
+	catchphrase = "R'CH T'H TR'TH"
+
+/obj/item/melee/touch_attack/mansus_fist/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
+	..()
+	if(ishuman(target))
+		var/mob/living/carbon/human/tar = target
+		if(tar.anti_magic_check())
+			tar.visible_message("<span class='danger'>Spell bounces off of [target]!</span>","<span class='danger'>The spell bounces off of you!</span>")
+			return
+	var/datum/mind/M = user.mind
+	var/datum/antagonist/ecult/cultie = M.has_antag_datum(/datum/antagonist/ecult)
+
+	for(var/X in cultie.get_all_knowledge())
+		var/datum/eldritch_knowledge/EK = X
+		EK.mansus_grasp_act(target, user, proximity_flag, click_parameters)
+
+	if(iscarbon(target))
```

Comment:
the way this is set up currently, it looks like you can whiff your mansus fist attack by clicking on the ground instead of the person standing on the ground, which basically means that you're screwed if you miss a single click against a moving target

Reply:
yeah but you are supposed to be able to click on anything, since mansus grasp is very modular, and some eldritch knowledge affects turfs.

Reply:
i agree it is a problem but i dont know how i could fix it, due to modularity of the spell

Reply:
you could just put the riders of the spell that require you to click on an object or a turf into a separate spell

in other words, make the bajillion object damage rider for the mansus hand that rust cultists get its own (touch) spell, then make mansus hands not try to expend their charge on objects/turfs

Reply:
Actually i think i know how what to do.